File: javascript/atbash-cipher/atbash-cipher.js
var alphabet = 'abcdefghijklmnopqrstuvwxyz1234567890'.split('');
var reverseAlpha = '0987654321abcdefghijklmnopqrstuvwxyz'.split('').reverse();

function Encode(input) {
  var inputArr = input.split('');
  var result = []
  for(var i = 0; i < inputArr.length; i++) {
    var index = alphabet.indexOf(inputArr[i].toLowerCase());
    result.push(reverseAlpha[index])
  }
  return result.join('')
}


module.exports = {
  encode: Encode
}

File: python/atbash-cipher/atbash_cipher.py
import re
plain  = 'abcdefghijklmnopqrstuvwxyz1234567890'
cipher = 'zyxwvutsrqponmlkjihgfedcba1234567890'


def encode(word):
    result = ''
    word = re.sub('[ .,]', '', word.lower())
    for letter in word:
        index = plain.find(letter)
        result += cipher[index]
    result = ' '.join([result[i:i+5] for i in range(0, len(result), 5)])
    return result

def decode(word):
    result = ''
    word = word.replace(' ', '')
    for letter in word:
        index = cipher.find(letter)
        result += plain[index]
    return result
